app/models/miq_worker.rb

```diff
@@ -205,31 +206,39 @@ def self.fetch_worker_settings_from_options_hash(options_hash, raw = false)
       classes.each do |c|
         section = section[c]
         raise _("Missing config section %{section_name}") % {:section_name => c} if section.nil?
+
         defaults = section[:defaults]
         unless defaults.nil?
-          defaults.delete_if {|k, v| v == Vmdb::Settings::RESET_VALUE }
+          defaults.delete_if { |_k, v| v == Vmdb::Settings::RESET_VALUE }
           settings.merge!(defaults)
         end
       end
 
-      section.delete_if {|k, v| v == Vmdb::Settings::RESET_VALUE }
+      section.delete_if { |_k, v| v == Vmdb::Settings::RESET_VALUE }
       settings.merge!(section)
 
       # If not specified, provide the worker_settings cleaned up in fixnums, etc. instead of 1.seconds, 10.megabytes
       # Clean up the configuration values in a format like "30.seconds"
-      unless raw == true
-        settings.keys.each do |k|
-          if settings[k].kind_of?(String)
-            if settings[k].number_with_method?
-              settings[k] = settings[k].to_i_with_method
-            elsif settings[k] =~ /\A\d+(.\d+)?\z/ # case where int/float saved as string
-              settings[k] = settings[k].to_i
-            end
-          end
+      normalize_settings!(settings) unless raw == true
+    end
+
+    settings
+  end
+
+  def self.normalize_settings!(settings)
+    settings.each_key do |k|
+      if settings[k].kind_of?(Hash)
+        normalize_settings!(settings[k])
+      elsif settings[k].kind_of?(String)
+        if settings[k].number_with_method?
+          settings[k] = settings[k].to_i_with_method
+        elsif settings[k].match?(/\A\d+(.\d+)?\z/) # case where int/float saved as string
+          settings[k] = settings[k].to_i
+        elsif ManageIQ::Password.encrypted?(settings[k])
+          settings[k] = ManageIQ::Password.decrypt(settings[k])
```

Comment:
NOTE I extracted this in one commit, and updated it to handle nested hashes in another commit if that makes it simpler to review